File: OrangeJuice.Server.Core.FSharp/Data/Logging/TraceErrorLog.fs
namespace OrangeJuice.Server.FSharp.Data.Logging

open System
open System.Collections
open System.Diagnostics

open Elmah

type TraceErrorLog() =
    inherit ErrorLog()

    static let traceSource : TraceSource = new TraceSource("TraceErrorLog")
    
    override this.GetError(id : string) =
        raise (new NotSupportedException())
    
    override this.GetErrors(pageIndex : int, pageSize : int, errorEntryList : System.Collections.IList) =
        raise (new NotSupportedException())
    
    override this.Log(error : Error) =
        let id = Guid.NewGuid().ToString()
        traceSource.TraceEvent(TraceEventType.Error, 1, "Error id={0}, message={1}, exception={2}", id, error.Message, error.Exception)
        id

File: OrangeJuice.Server.Core/Data/Logging/TraceErrorLog.cs
using System;
using System.Collections;
using System.Diagnostics;

using Elmah;

namespace OrangeJuice.Server.Data.Logging
{
	public class TraceErrorLog : ErrorLog
	{
		private static readonly TraceSource _traceSource = new TraceSource("TraceErrorLog");

		#region ErrorLog members
		public override string Log(Error error)
		{
			string id = Guid.NewGuid().ToString();
			
			_traceSource.TraceEvent(TraceEventType.Error, 1, "Error id={0}, message={1}, exception={2}", id, error.Message, error.Exception);

			return id;
		}

		public override ErrorLogEntry GetError(string id)
		{
			throw new NotSupportedException();
		}

		public override int GetErrors(int pageIndex, int pageSize, IList errorEntryList)
		{
			throw new NotSupportedException();
		}
		#endregion
	}
}